examples/solids/qfunctions/hyperFS.h

```diff
@@ -0,0 +1,360 @@
+    const CeedScalar Fa11 = F[0][0]*F[2][2] - F[0][2]*F[2][0];
+    const CeedScalar Fa12 = F[0][2]*F[1][0] - F[0][0]*F[1][2];
+    const CeedScalar Fa20 = F[1][0]*F[2][1] - F[1][1]*F[2][0];
+    const CeedScalar Fa21 = F[0][1]*F[2][0] - F[0][0]*F[2][1];
+    const CeedScalar Fa22 = F[0][0]*F[1][1] - F[0][1]*F[1][0];
+    const CeedScalar detF = F[0][0]*Fa00 + F[1][0]*Fa01 + F[2][0]*Fa02;
+
+    // *INDENT-OFF*
+    const CeedScalar F_inv[3][3] = {{Fa00/detF, Fa01/detF, Fa02/detF},
+                                    {Fa10/detF, Fa11/detF, Fa12/detF},
+                                    {Fa20/detF, Fa21/detF, Fa22/detF}
+                                   };
+    // *INDENT-ON*
+
+    // C : right Cauchy-Green tensor
+    // C = F^T * F (^T  means Transpose)
+    CeedScalar C[3][3];
+    for (int j = 0; j < 3; j++)
+        for (int k = 0; k < 3; k++) {
+            C[j][k] = 0;
+            for(int m = 0; m < 3; m++)
+               C[j][k] += F[k][m] * F[j][m];
+        }
+
+
+    // Compute C^(-1) : C-Inverse
+    const CeedScalar A00 = C[1][1]*C[2][2] - C[1][2]*C[2][1];
+    const CeedScalar A01 = C[0][2]*F[2][1] - C[0][1]*C[2][2];
+    const CeedScalar A02 = C[0][1]*C[1][2] - C[0][2]*C[1][1];
+    const CeedScalar A10 = C[1][2]*C[2][0] - C[1][0]*C[2][2];
+    const CeedScalar A11 = C[0][0]*C[2][2] - C[0][2]*C[2][0];
+    const CeedScalar A12 = C[0][2]*C[1][0] - C[0][0]*C[1][2];
+    const CeedScalar A20 = C[1][0]*C[2][1] - C[1][1]*C[2][0];
+    const CeedScalar A21 = C[0][1]*C[2][0] - C[0][0]*C[2][1];
+    const CeedScalar A22 = C[0][0]*C[1][1] - C[0][1]*C[1][0];
+    const CeedScalar detC = C[0][0]*A00 + C[1][0]*A01 + C[2][0]*A02;
+
+    // *INDENT-OFF*
+    const CeedScalar C_inv[3][3] = {{A00/detC, A01/detC, A02/detC},
+                                    {A10/detC, A11/detC, A12/detC},
+                                    {A20/detC, A21/detC, A22/detC}
+                                   };
+
+    // Compute the Second Piola-Kirchhoff (S)
+    const CeedScalar llnj_m = lambda *log(detF)-mu;
+    CeedScalar S[3][3] = {{mu + llnj_m*C_inv[0][0],      llnj_m*C_inv[0][1],      llnj_m*C_inv[0][2]},
+                          {     llnj_m*C_inv[1][0], mu + llnj_m*C_inv[1][1],      llnj_m*C_inv[1][2]},
+                          {     llnj_m*C_inv[2][0],      llnj_m*C_inv[2][1], mu + llnj_m*C_inv[2][2]}
+                         };
+    // *INDENT-ON*
+
+    // dPdF is indexed as an array of 81  elements
+    CeedScalar dPdF[81];
+    for (int j = 0; j < 3; j++)
+      for(int J = 0; J < 3; J++)
+        for(int a = 0; a < 3; a++)
+          for(int A = 0; A < 3; A++)
+            dPdF[27*j+9*J+3*a+A] = (a == j ? 1 : 0) * (S[A][J]) +
+                                   lambda*F_inv[A][a]*F_inv[J][j]
+                                   - (lambda*log(detF) - mu) *
+                                   (F_inv[A][j]*F_inv[J][a] +
+                                   (a == j ? 1 : 0) * C_inv[A][J]);
```

Comment:
This is my biggest efficiency concern, and is very unnecessary. Removing this should also reduce register pressure (relevant to both CPU and GPU). Cc: @jeremylt

Reply:
To clarify, do you mean that with your recommended formulation we can avoid constructing this full tensor or that there is a better way to fill this tensor?

Reply:
No tensor. Did the explanation in my docs make sense?

Reply:
Yeah. I think I swapped over, but now I've lost convergence, so something has to be wrong.

Reply:
Now it seems to be fine.